project-base/app/src/DataFixtures/Demo/DemoDataFactory/ProductDemoDataFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataFixtures\Demo\DemoDataFactory;
+
+use App\DataFixtures\Demo\UnitDataFixture;
+use App\DataFixtures\Demo\VatDataFixture;
+use App\Model\Product\ProductData;
+use App\Model\Product\ProductDataFactory;
+use App\Model\Product\Unit\Unit;
+use DateTime;
+use Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat;
+
+class ProductDemoDataFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \App\Model\Product\ProductDataFactory $productDataFactory
+     * @param \Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade $persistentReferenceFacade
+     */
+    public function __construct(
+        private readonly Domain $domain,
+        private readonly ProductDataFactory $productDataFactory,
+        private readonly PersistentReferenceFacade $persistentReferenceFacade,
+    ) {
+    }
+
+    /**
+     * @return \App\Model\Product\ProductData
+     */
+    public function createDefaultData(): ProductData
+    {
+        $productData = $this->productDataFactory->create();
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setSellingTo($productData, null);
```

Comment:
is it necessary to call this method here? the null values should be initialized already after creating the data object by `ProductDataFactory`. So I believe all the `set*` methods can have non-nullable their second argument - if you want to set a particular value to the field, call the setter, if you want to keep it null, do not do anything :shrug: In this manner, we can get rid of multiple lines in `ProductDataFixture` as well - e.g. `$this->setBrand($productData, null);` is called 75 times for vain, `$this->setFlags($productData, []);` is called 99 times in vain.

Reply:
you are right, removed all unnecessary ballast i saw